### ISSUE TITLE
Some pipeline steps cannot run in parallel when datadog plugin is installed

### ISSUE THREAD

**Issue by jesperes** [jenkinsci/datadog-plugin#251]
Some pipeline steps, e.g. `rtp` cannot run in parallel when the datadog plugin is installed. This triggers a long stacktrace (see below)"

```
java.util.ConcurrentModificationException
		at java.util.HashMap$HashIterator.nextNode(HashMap.java:1445)
		at java.util.HashMap$EntryIterator.next(HashMap.java:1479)
		at java.util.HashMap$EntryIterator.next(HashMap.java:1477)
		at com.thoughtworks.xstream.converters.collections.MapConverter.marshal(MapConverter.java:75)
		at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
		at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
		at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:83)
		at hudson.util.RobustReflectionConverter.marshallField(RobustReflectionConverter.java:275)
		at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:262)
	Caused: java.lang.RuntimeException: Failed to serialize org.datadog.jenkins.plugins.datadog.traces.StepDataAction#stepDataByDescriptor for class org.datadog.jenkins.plugins.datadog.traces.StepDataAction
		at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:266)
		at hudson.util.RobustReflectionConverter$2.visit(RobustReflectionConverter.java:233)
		at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:150)
		at hudson.util.RobustReflectionConverter.doMarshal(RobustReflectionConverter.java:219)```
...
```

**To Reproduce**

Run the following pipeline in a Jenkins instance with the datadog plugin installed:

```groovy
node("medium") {
    Map jobs = [:]
    1.upto(100) { i ->
        jobs[i] = {
            rtp(nullAction: '1', stableText: "${i}")
        }
    }
    parallel(jobs)
}
```

**Expected behavior**

The `rtp` pipeline step should be executed 100 times with the text `${i}`.

**Environment and Versions:**

* Jenkins 2.289.3
* Datadog plugin 3.1.1

**Comment by jesperes** [comment]
Also: I tried adding the job to the `Excluded Jobs` section, but that did not have any effect.

**Comment by drodriguezhdez** [comment]
Hi @jesperes,

This bug should be fixed in this PR #243, available in the 3.1.2+ versions 👌🏼

**Comment by jesperes** [comment]
Thanks for the quick reply.
